### PR TITLE
creation of README.MD

### PR DESCRIPTION
**Pull request by ikonia**
after having a minor problem using the pulp plugin with foreman on a remote pulp host I realised the functionality I was working on was never going to work / supported. This could have been mitigated with a basic README type document in the repo to give some basic overview. This is a first draft that can be expanded on by people, but it provides some base coverage, dependencies and config example.

### CODE REVIEW

**Comment by Areyus** [comment]
Since this repo is currently the only place to have any documentation on smart-proxy plugins, I agree having at least a basic readme is a good idea. Additional +1 for pointing out that this will only work in a Katello setup.
Maybe someone with better knowledge of the workings of this plugin can add some more details.

File: README.MD [hunk 1]
```diff
@@ -0,0 +1,68 @@
+#  Foreman Pulp Plugin
+
+Foreman project plugin for Pulp allowing Katello hosts to interact with Pulp application services for content management
+While this plugin is part of the Foreman project it can only be used with Katello as Foreman is not content aware without the Katello plugin.
+
+## Getting Started
+
+The Foreman project provides documentation on this specific plugin installation from multiple methods, distribution specific RPM, distribution specific DEB and direct manual build from source. 
+The plugin can also be installed as part of the foreman-installer puppet class with the "--[no-]enable-foreman-proxy-plugin-Pulp" option.
```
Comment (ekohl) [inline README.MD]: Pulp -> pulp in the installer option.

File: README.MD [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+Documentation can be found here for how to install a stand alone smart proxy https://theForeman.org/manuals/$current_foreman_version/index.html#4.3.1SmartProxyInstallation
```
Comment (ekohl) [inline README.MD]: We have a `latest` redirect instead of `$current_foreman_version`.

File: README.MD [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+### Prerequisites
+
+A working Katello instance (note not Foreman - must be Katello)
+A smart proxy or capsule with the Pulp plugin installed and enabled
+A Pulp installation as part of the Katello install, or a remote Pulp service to connect to
+
+
+### Installing
+
+Assuming prerequisites are met, the Pulp plugin must be enabled in the Foreman-proxy directory with the Pulp.yaml and Pulpnode.yaml
```
Comment (ekohl) [inline README.MD]: The filenames should be lowercase.

File: README.MD [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+the following parameters are required to be set
+
+```
```
Comment (ekohl) [inline README.MD]: You can add yaml here for syntax highlighting. Same in the example below.

File: README.MD [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+## Running the tests
+
+Todo - some test examples
+
+
+## Authors
+
+* **Matt Darcy** - *Initial Draft* - [ikonia](https://github.com/ikonia)
```
Comment (ekohl) [inline README.MD]: This reads as if you're the author of the project, not the README itself.

**Comment by ikonia** [comment]
changes updated in line with feedback, should hopefully be acceptable now.

File: README.MD (hunk 5 above)
Comment (ikonia) [reply README.MD]: changes made in line with feedback credit given to foreman project properly

File: README.MD (hunk 4 above)
Comment (ikonia) [reply README.MD]: yaml syntax support included

File: README.MD (hunk 3 above)
Comment (ikonia) [reply README.MD]: files correctly cased.

File: README.MD (hunk 2 above)
Comment (ikonia) [reply README.MD]: updated and tested URL, resolves correctly, useful redirect.

File: README.MD (hunk 1 above)
Comment (ikonia) [reply README.MD]: now fixed this to correct lower case option, missed this when fixing other casing issues.

File: README.MD [hunk 6]
```diff
@@ -0,0 +1,70 @@
+#  Foreman Pulp Plugin
+
+Foreman project plugin for Pulp allowing Katello hosts to interact with Pulp application services for content management
+While this plugin is part of the Foreman project it can only be used with Katello as Foreman is not content aware without the Katello plugin.
+
+## Getting Started
+
+The Foreman project provides documentation on this specific plugin installation from multiple methods, distribution specific RPM, distribution specific DEB and direct manual build from source. 
+The plugin can also be installed as part of the foreman-installer puppet class with the "--[no-]enable-foreman-proxy-plugin-pulp" option.
+Documentation can be found here for how to install a stand alone smart proxy https://theForeman.org/manuals/latest/index.html#4.3.1SmartProxyInstallation
```
Comment (ekohl) [inline README.MD]: theforeman.org

File: README.MD [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+### Prerequisites
+
+A working Katello instance (note not Foreman - must be Katello)
```
Comment (ekohl) [inline README.MD]: If you use `* ` you get bullet points and it's easier to read

Comment (ikonia) [reply README.MD]: changed prereqs to bullet points, cleaner read

File: README.MD (hunk 6 above)
Comment (ikonia) [reply README.MD]: fixed case error in the URL, excellent spot.

**Comment by ikonia** [comment]
as you say a good place to start, I'll actually get some test cases written shorty to update this and critically (which is what started my original work) running this on a remote pulp server wrapped with a capsule / smart proxy, I'll get that use case documented too. 
Other feedback welcome to get this in the best shape before merged

**Comment by ekohl** [comment]
Thanks!

**Pull request merged**